### ISSUE TITLE
[opencv] build failure

### ISSUE THREAD

**Issue by cyh19910216** [microsoft/vcpkg#38321]
### Operating system

Windows 11 23H2 x64

### Compiler

Visual Studio 2022 17.9.6

### Steps to reproduce the behavior

```Shell
.\vcpkg install opencv[contrib,cuda,cudnn,eigen,ffmpeg,opengl,openmp]:x64-windows
```

### Failure logs

PS D:\Data\vcpkg> .\vcpkg install opencv[contrib,cuda,cudnn,eigen,ffmpeg,opengl,openmp]:x64-windows
Computing installation plan...
The following packages will be built and installed:
    opencv[contrib,core,cuda,cudnn,default-features,eigen,ffmpeg,opengl,openmp]:x64-windows@4.8.0#1
  * opencv4[contrib,core,cuda,cudnn,default-features,dnn,eigen,ffmpeg,jpeg,opengl,openmp,png,quirc,tiff,webp]:x64-windows@4.8.0#15
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
-- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:10809".
Compiler found: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.39.33519/bin/Hostx64/x64/cl.exe
Restored 0 package(s) from C:\Users\CYH\AppData\Local\vcpkg\archives in 1.27 ms. Use --debug to see more details.
Installing 1/2 opencv4[contrib,core,cuda,cudnn,default-features,dnn,eigen,ffmpeg,jpeg,opengl,openmp,png,quirc,tiff,webp]:x64-windows@4.8.0#15...
Building opencv4[contrib,core,cuda,cudnn,default-features,dnn,eigen,ffmpeg,jpeg,opengl,openmp,png,quirc,tiff,webp]:x64-windows@4.8.0#15...
-- Using cached opencv4-e5e1a3bfdea96bebda2ad963bc8f6cf17930aef7.patch.
-- Using cached opencv-opencv-4.8.0.tar.gz.
-- Cleaning sources at D:/Data/vcpkg/buildtrees/opencv4/src/4.8.0-2bf495557d.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/Data/vcpkg/downloads/opencv-opencv-4.8.0.tar.gz
-- Applying patch 0001-disable-downloading.patch
-- Applying patch 0002-install-options.patch
-- Applying patch 0003-force-package-requirements.patch
-- Applying patch 0004-fix-eigen.patch
-- Applying patch 0005-fix-policy-CMP0057.patch
-- Applying patch 0006-fix-uwp.patch
-- Applying patch 0008-devendor-quirc.patch
-- Applying patch 0009-fix-protobuf.patch
-- Applying patch 0010-fix-uwp-tiff-imgcodecs.patch
-- Applying patch 0011-remove-python2.patch
-- Applying patch 0012-fix-zlib.patch
-- Applying patch 0015-fix-freetype.patch
-- Applying patch 0017-fix-flatbuffers.patch
-- Applying patch 0019-missing-include.patch
-- Applying patch 0020-fix-compat-cuda12.2.patch
-- Applying patch 0021-static-openvino.patch
-- Applying patch D:/Data/vcpkg/downloads/opencv4-e5e1a3bfdea96bebda2ad963bc8f6cf17930aef7.patch
-- Applying patch 0022-fix-supportqnx.patch
-- Using source at D:/Data/vcpkg/buildtrees/opencv4/src/4.8.0-2bf495557d.clean
-- Using cached opencv-cache/tiny_dnn/adb1c512e09ca2c7a6faef36f9c53e59-v1.0.0a3.tar.gz.
-- Using cached opencv-cache/nvidia_optical_flow/a73cd48b18dcc0cc8933b30796074191-edb50da3cf849840d680249aa6dbef248ebce2ca.zip.
-- Using cached opencv-opencv_contrib-4.8.0.tar.gz.
-- Cleaning sources at D:/Data/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/Data/vcpkg/downloads/opencv-opencv_contrib-4.8.0.tar.gz
-- Applying patch 0007-fix-hdf5.patch
-- Applying patch 0016-fix-freetype-contrib.patch
-- Applying patch 0018-fix-depend-tesseract.patch
-- Applying patch 0019-fix-ogre-dependency.patch
-- Using source at D:/Data/vcpkg/buildtrees/opencv4/src/4.8.0-8d756cdf2d.clean
-- Using cached opencv-cache/wechat_qrcode/238e2b2d6f3c18d6c3a30de0c31e23cf-detect.caffemodel.
-- Using cached opencv-cache/wechat_qrcode/cbfcd60361a73beb8c583eea7e8e6664-sr.caffemodel.
-- Using cached opencv-cache/wechat_qrcode/6fb4976b32695f9f5c6305c19f12537d-detect.prototxt.
-- Using cached opencv-cache/wechat_qrcode/69db99927a70df953b471daaba03fbef-sr.prototxt.
-- Using cached opencv-cache/xfeatures2d/boostdesc/0ea90e7a8f3f7876d450e4149c97c74f-boostdesc_bgm.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/232c966b13651bd0e46a1497b0852191-boostdesc_bgm_bi.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/324426a24fa56ad9c5b8e3e0b3e5303e-boostdesc_bgm_hd.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/202e1b3e9fec871b04da31f7f016679f-boostdesc_binboost_064.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/98ea99d399965c03d555cef3ea502a0b-boostdesc_binboost_128.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/e6dcfa9f647779eb1ce446a8d759b6ea-boostdesc_binboost_256.i.
-- Using cached opencv-cache/xfeatures2d/boostdesc/0ae0675534aa318d9668f2a179c2a052-boostdesc_lbgm.i.
-- Using cached opencv-cache/xfeatures2d/vgg/e8d0dcd54d1bcfdc29203d011a797179-vgg_generated_48.i.
-- Using cached opencv-cache/xfeatures2d/vgg/7126a5d9a8884ebca5aea5d63d677225-vgg_generated_64.i.
-- Using cached opencv-cache/xfeatures2d/vgg/7cd47228edec52b6d82f46511af325c5-vgg_generated_80.i.
-- Using cached opencv-cache/xfeatures2d/vgg/151805e03568c9f490a5e3a872777b75-vgg_generated_120.i.
-- Using cached opencv-cache/data/7505c44ca4eb54b4ab1e4777cb96ac05-face_landmark_model.dat.
-- Found external ninja('1.11.0').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: D:/Data/vcpkg/downloads/tools/cmake-3.29.0-windows/cmake-3.29.0-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j9
    Working Directory: D:/Data/vcpkg/buildtrees/opencv4/x64-windows-dbg
    See logs for more information:
      D:\Data\vcpkg\buildtrees\opencv4\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/opencv4/portfile.cmake:479 (vcpkg_cmake_install)
  scripts/ports.cmake:175 (include)

error: building opencv4:x64-windows failed with: BUILD_FAILED
Elapsed time to handle opencv4:x64-windows: 18 min
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+opencv4
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[opencv4]+Build+error+on+x64-windows&body=Copy+issue+body+from+D%3A%2FData%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/15060688/install-x64-windows-dbg-out.log)

**Comment by cyh19910216** [comment]
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/15060749/install-x64-windows-dbg-out.log)

**Comment by JonLiu1993** [comment]
@cyh19910216, I can not reproduce your error on my local machine:
````
Elapsed time to handle opencv:x64-windows: 765 ms
Total install time: 3.1 h
opencv provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(OpenCV CONFIG REQUIRED)
  # note: 59 additional targets are not displayed.
  target_link_libraries(main PRIVATE opencv_ml opencv_dnn opencv_dpm opencv_hdf)

````
Please make sure you are using the latest port files with git pull and vcpkg update. Delete all opencv caches and try again.

**Comment by cyh19910216** [comment]
> @cyh19910216, I can not reproduce your error on my local machine:@cyh19910216，我无法在本地计算机上重现您的错误：
> 
> ```
> Elapsed time to handle opencv:x64-windows: 765 ms
> Total install time: 3.1 h
> opencv provides CMake targets:
> 
>   # this is heuristically generated, and may not be correct
>   find_package(OpenCV CONFIG REQUIRED)
>   # note: 59 additional targets are not displayed.
>   target_link_libraries(main PRIVATE opencv_ml opencv_dnn opencv_dpm opencv_hdf)
> ```
> 
> Please make sure you are using the latest port files with git pull and vcpkg update. Delete all opencv caches and try again.

Thank you! But I fail again.
I have used "git pull" and "vcpkg update", and deleted the folders "opencv", "opencv4", "opencv-cache" and "temp".
But it fails again.

CPU: Intel Core i5-1155G7
GPU: NVIDIA GeForce MX450
CUDA: 12.4

**Comment by burning-star-x** [comment]
The latest version of OpenCV available in vcpkg is 4.8.0.

OpenCV version 4.8.0 supports the latest CUDA version, which is 12.0.

so, use CUDA  12.0
